go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -70,21 +74,64 @@ type serviceConfig struct {
 	service *kapi.Service
 	// Were those rules etp:local + Host Networked rules
 	etpHostRules bool
+	// Used when svc=etp:local + it has non local host networked backends in LGW.
+	// NOTE: For most part this is always the opposite of etpHostRules, however there
+	// can be a situation where there are no endpoints for a service in which case
+	// both etpHostRules & etpNonHostRules will be false.
+	etpNonHostRules bool
 }
 
-// updateServiceFlowCache handles managing shared gateway flows for ingress traffic towards kubernetes services
+// createETPLocalNonHostOFRules adds the necessary openflows for a service of type ETP=local when its backed by non-local-host-networked endpoints
+// cookie is the hash value of openflow cookie,  svcPort is the nodePort of the service, flowProtocol is either v4 or v6,
+// ipaddress is either externalIP or LB.status.ingress, nwDst&nwSrc are constants used to match on the destinations based on the flowprotocol
+func (npw *nodePortWatcher) createETPLocalNonHostOFRules(cookie string, svcPort int32, flowProtocol string, ipAddress, nwDst, nwSrc string) []string {
+	var nodeportFlowsResult []string
+	if len(ipAddress) != 0 && len(nwDst) != 0 && len(nwSrc) != 0 {
+		// LB/EIP
+		nodeportFlowsResult = append(nodeportFlowsResult,
+			fmt.Sprintf("cookie=%s, priority=110, in_port=%s, %s, %s=%s, tp_dst=%d, actions=ct(commit,zone=%d,table=6)",
+				cookie, npw.ofportPhys, flowProtocol, nwDst, ipAddress, svcPort, HostNodePortCTZone),
+			fmt.Sprintf("cookie=%s, priority=110, in_port=LOCAL, %s, %s=%s, tp_src=%d, actions=ct(zone=%d,table=7)",
+				cookie, flowProtocol, nwSrc, ipAddress, svcPort, HostNodePortCTZone))
+	} else {
+		// NP
+		nodeportFlowsResult = append(nodeportFlowsResult,
+			// Traffic destined for ETP=local backed by OVN-K pod in LGW mode is matched on the svcNP and sent to table6.
+			fmt.Sprintf("cookie=%s, priority=110, in_port=%s, %s, tp_dst=%d, actions=ct(commit,zone=%d,table=6)",
```

Comment:
wouldnt this only be the case in local gateway mode? in shared it should go to OVN right? Is this function only for local gw mode or what am I missing?

Reply:
correct this function is only called for LGW.